tests/testtransport.nim

```diff
@@ -118,16 +120,19 @@ suite "TCP transport":
   test "e2e: handle read":
     proc testListenerDialer(): Future[bool] {.async.} =
       let ma: MultiAddress = Multiaddress.init("/ip4/0.0.0.0/tcp/0")
+      var finished = newAsyncEvent()
       proc connHandler(conn: Connection): Future[void] {.async, gcsafe.} =
         let msg = await conn.read(6)
         check cast[string](msg) == "Hello!"
+        finished.fire()
 
       let transport1: TcpTransport = newTransport(TcpTransport)
       asyncCheck await transport1.listen(ma, connHandler)
 
       let transport2: TcpTransport = newTransport(TcpTransport)
       let conn = await transport2.dial(transport1.ma)
       await conn.write(cstring("Hello!"), 6)
+      await finished.wait()
```

Comment:
`transport2` is leaking here.

Reply:
again, I can't see how it leaks, can you please elaborate?

Reply:
confirmed leak :) I close in my PR too